create_bot_knowledge.py

```diff
@@ -80,139 +80,107 @@ def connect_to_source(project: Tuple[str, str]):
     return service, repo
 
 
-def extract_knowledge_from_repository(project: Tuple[str, str], update_knowledge: bool = False):
+def check_directory(knowledge_dir):
+    if not knowledge_dir.exists():
+        _LOGGER.info("No knowledge from any repo has ever been created, creating new directory at %s" % knowledge_dir)
+        os.mkdir(knowledge_dir)
+
+def is_old_knowledge(project_knowledge):
+    if project_knowledge.exists():
+        _LOGGER.info("Previous collected knowledge of repo %s/%s found" % (project[1], project[0]))
+        return True
+    
+    _LOGGER.info("No previous knowledge from repo %s/%s found" % (project[1], project[0]),
+                 "New knowledge file will be created")
+
+
+def pull_analysis(pull, results):
```

Comment:
https://github.com/packit-service/ogr/blob/32c567e6d7f5de83f12bbba7852031b493a91c63/ogr/services/github/project.py#L241
add at the beginning of file `from ogr.services.github.pull_request import GithubPullRequest`

`def pull_analysis(pull: GithubPullRequest, results: Dict[str, Dict[str, Optional[str, float])`

Reply:
soon changes with type hints will be committed, in this particular case i've used:
```
pull_analysis(pull: PullRequest, results: Dict[str, Dict[str, Union[Optional[str], float]]])
```
(the PullRequest type is already imported)

Reply:
the reason i've used `Union[Optional[str], float]` is that the TTR field must always have some value (float), but the other fields can be either str or None (therefore Optional[str])

Reply:
Great :) Thanks @xtuchyna